crates/uv-toolchain/src/implementation.rs

```diff
@@ -30,10 +30,37 @@ impl ImplementationName {
         NAMES.iter()
     }
 
-    pub fn as_str(self) -> &'static str {
+    pub fn pretty(self) -> &'static str {
```

Comment:
Is there a standard pattern or name for this?

Reply:
I guess the capitalized version is the "display name" but much more often want the lowercase string, should I be doing lowercase casts all over the place instead?

Reply:
Hmmmmmmmmm.

Usually you'd have a `Display` impl here somewhere, but given that you can return a `&'static str`, that's usually a bit nicer to work with.

If there were only one sensible string representation, then I think I'd call this `as_str`.

If there is a sensible "default" representation that should be used in almost all cases and another representation that is used less frequently, then I think I'd call the default `as_str` and the other something more descriptive, like, `as_str_capitalized` or something.

If there is no obvious default, I think I'd just be descriptive with both names. Like, `as_str_lowercase` and `as_str_capitalized`. Or something like that.

With that said, I don't mind `pretty` either.

Something bothers me about having one form be a concrete method and another form be a `From` impl. I can't quite articulate it, but it seems a little confusing. Like, it would be easy to see one but not the other. You could add concrete methods for both, and then add a `From` impl with whatever the "default" is I guess.

I think overall the right thing to do here depends a lot on how these are used and whether there's a clear default or whatever.

Reply:
Thanks for all the context! We do have a `Display` representation which is the lowercase name right now. I used to have `as_str` for the capitalized name but changed it. Maybe I'll change it back... I think it's probably better to consider outside this pull request though.